politeiad/cmd/legacypoliteia/convertcmd.go

```diff
@@ -0,0 +1,451 @@
+	"github.com/google/uuid"
+)
+
+const (
+	// Default command settings
+	defaultLegacyDir = "./legacy-politeia-data"
+
+	// filePermissions is the file permissions that are used for all directory
+	// and file creation in this tool.
+	filePermissions = 0755
+)
+
+var (
+	// CLI flags for the convert command
+	convertFlags = flag.NewFlagSet(convertCmdName, flag.ContinueOnError)
+	legacyDir    = convertFlags.String("legacydir", defaultLegacyDir,
+		"default legacy data dir")
+	skipComments = convertFlags.Bool("skipcomments", false, "skip comments")
+	skipBallots  = convertFlags.Bool("skipballots", false, "skip ballots")
+	ballotLimit  = convertFlags.Int("ballotlimit", 0, "limit parsed votes")
+	userID       = convertFlags.String("userid", "", "replace user IDs")
+)
+
+type convertCmd struct {
+	client       *http.Client
+	gitRepo      string
+	legacyDir    string
+	skipComments bool
+	skipBallots  bool
+	ballotLimit  int
+	userID       string
+}
+
+// execConvertComd executes the convert command.
+//
+// The convert command parses a legacy git repo, converts the data into types
+// supported by the tstore backend, then writes the converted JSON data to
+// disk. This data can be imported into tstore using the 'import' command.
+func execConvertCmd(args []string) error {
+	// Verify the git repo exists
+	if len(args) == 0 {
+		return fmt.Errorf("missing git repo argument")
+	}
+	gitRepo := util.CleanAndExpandPath(args[0])
+	if _, err := os.Stat(gitRepo); err != nil {
+		return fmt.Errorf("git repo not found: %v", gitRepo)
+	}
+
+	// Parse the CLI flags
+	err := convertFlags.Parse(args[1:])
+	if err != nil {
+		return err
+	}
+
+	// Clean the legacy directory path
+	*legacyDir = util.CleanAndExpandPath(*legacyDir)
+
+	// Verify the user ID
+	if *userID != "" {
+		_, err = uuid.Parse(*userID)
+		if err != nil {
+			return fmt.Errorf("invalid user id '%v': %v", *userID, err)
+		}
+	}
+
+	// Setup the legacy directory
+	err = os.MkdirAll(*legacyDir, filePermissions)
+	if err != nil {
+		return err
+	}
+
+	client, err := util.NewHTTPClient(false, "")
+	if err != nil {
+		return err
+	}
+
+	// Setup the cmd context
+	c := convertCmd{
+		client:       client,
+		gitRepo:      gitRepo,
+		legacyDir:    *legacyDir,
+		skipComments: *skipComments,
+		skipBallots:  *skipBallots,
+		ballotLimit:  *ballotLimit,
+		userID:       *userID,
+	}
+
+	// Convert the git proposals
+	return c.convertGitProposals()
+}
+
+// convertGitProposals converts the git proposals to tstore proposals, saving
+// the tstore proposals to disk as the conversion is finished.
+func (c *convertCmd) convertGitProposals() error {
+	// Build an inventory of all git proposal tokens
+	tokens, err := gitProposalTokens(c.gitRepo)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Found %v legacy git proposals\n", len(tokens))
+
+	// Parse git vote timestamps using git log. If parsed ballot is limited
+	// skip this.
+	var ts map[string]map[string]int64
+	if c.ballotLimit == 0 {
```

Comment:
This means that if ballotLimit=10, those 10 votes will not have timestamps.  This is incorrect behavior.  Why is it like this?